content/en/blog/2024/inpod-traffic-redirection-ambient/index.md

```diff
@@ -0,0 +1,229 @@
+with various Kubernetes platforms with different Container Network Interface(CNI) layers. After all,
+Ambient wouldn’t be ambient without being able to run everywhere with any CNIs.
+
+At Solo, we had been working hard for 9 months on an innovative internal solution to solve this and we had decided
+to [upstream](https://github.com/istio/istio/issues/48212) our changes in late 2023 to help Ambient reach beta faster
+so more users can operate ambient in Istio 1.21 or newer and enjoy the benefits of ambient sidecar-less mesh in their
+platforms regardless of their preferred CNIs.
+
+## How do we get here?
+
+### Istio and CNI - What’s The Relationship?
+
+Istio is a service mesh, and service meshes by definition are not *CNI implementations* - service meshes build on top of the
+low-level, standards-compliant CNI implementation present in Kubernetes clusters to offer the high-level policy and security
+controls users need at scale.
+
+Basically, before you can do things like secure pod traffic with mTLS and apply lots of high-level authentication and authorization policy
+at the service mesh layer, you must have a functional Kubernetes cluster with a functional CNI implementation, to make sure the
+basic networking pathways are set up so that packets can get from one pod to another (and from one node to another) in
+your cluster.
+
+Though some service meshes may also ship and require their own in-house CNI implementation, the Istio project has chosen not
+to, for maximum compatibility with managed offerings, cross-vendor support, and composability with the broader CNCF ecosystem.
+
+### Traffic redirection in ambient alpha
+
+The [Istio-cni](/docs/setup/additional-setup/cni/) component is an optional component in the sidecar data plane mode,
+commonly used to remove the [requirement for the `NET_ADMIN` and `NET_RAW` capabilities](/docs/ops/deployment/requirements/) for
+users deploying pods into the Istio mesh. With the launch of ambient, istio-cni has become a required component for ambient
+data plane mode. Whenever pods are added to ambient mesh, the istio-cni component configures traffic redirection for all
+incoming and outgoing traffic between the application pods and their co-located [ztunnel](/blog/2023/rust-based-ztunnel/) on
+the node via the node-level network namespace. The key difference between sidecar and ambient on the traffic redirection is
+the redirection in ambient is across network namespaces, from the application namespace to the Istio’s system namespace
+where ztunnel typically runs.
+
+As we tested more broadly in multiple Kubernetes environments which many of them have its default CNI, it became clear that
+the host network based traffic redirection mechanism we had been using for Istio ambient traffic capture and redirection
+during the alpha stage was not going to meet our requirements. The following challenges were identified:
```

Comment:
Did we not know this in advance?
"which many of them have its default CNI" - which default CNI?  The default Kubernetes CNI?  Is there such a thing, or do you mean the default CNI for an environment?

Reply:
i.e. Do you mean "many of which have their own default CNI"?

Reply:
We should consistently say _CNI implementation_ everywhere we are not talking about the Istio component, I think.